core_v2/sources/domains.move

```diff
@@ -505,14 +527,124 @@ module aptos_names_v2::domains {
         domain_name: String
     ) acquires CollectionCapabilityV2, NameRecordV2 {
         config::assert_signer_is_admin(sign);
-        let record = get_record_mut(domain_name, subdomain_name);
+        let record = get_record_mut_deprecated(domain_name, subdomain_name);
         object::transfer_with_ref(
             object::generate_linear_transfer_ref(&record.transfer_ref),
             get_token_signer_address()
         );
         record.target_address = option::none();
     }
 
+    public fun renew_domain(
+        sign: &signer,
+        domain_name: String,
+        num_years: u8,
+    ) acquires CollectionCapabilityV2, NameRecord, RegisterNameEventsV1 {
+        // check the domain eligibility
+        let (is_valid, length) = utf8_utils::string_is_allowed(&domain_name);
+        assert!(is_valid, error::invalid_argument(EDOMAIN_HAS_INVALID_CHARACTERS));
+        assert!(length <= config::max_domain_length(), error::out_of_range(EDOMAIN_TOO_LONG));
+        assert!(length >= config::min_domain_length(), error::out_of_range(EDOMAIN_TOO_SHORT));
+
+        let price = price_model::price_for_domain_v1(length, num_years);
+        renew_domain_internal(sign, domain_name, num_years, price);
+    }
+
+    fun renew_domain_internal(
```

Comment:
I pass in price here so that we can call this func directly when we want to force renewal without paying price.